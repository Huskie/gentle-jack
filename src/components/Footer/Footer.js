import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'react-emotion';
import { withTheme } from 'emotion-theming';

import { Address, Container, KeyValuePair } from '../';

const StyledFooter = Styled('footer')`
    background: ${props => props.theme.colours.secondary};
    bottom: 0;
    color: #aaa;
    padding: 2rem 0;
    position: fixed;
    width: 100%;
`;

const BaseFooter = () => (
    <StyledFooter>
        <Container>
            <Address>
                <span>Registered office address:</span>
                <span>272 Bath Street</span>
                <span>Glasgow, G2 4JR</span>
            </Address>
            <KeyValuePair theKey="Company number" value="SC588866" />
            <KeyValuePair theKey="VAT registration number" value="XXXXXXX" />
        </Container>
    </StyledFooter>
);

const Footer = withTheme(BaseFooter);

Footer.defaultProps = {};

Footer.propTypes = {};

export { Footer };
