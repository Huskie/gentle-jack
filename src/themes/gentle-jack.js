const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
