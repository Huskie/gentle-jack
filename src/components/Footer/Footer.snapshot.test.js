import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { Footer } from './Footer';
import { theme } from '../../themes/gentle-jack';

describe('Footer snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
