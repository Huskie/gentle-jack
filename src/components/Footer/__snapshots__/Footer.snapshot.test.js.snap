// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshots: component renders with defaults 1`] = `
.emotion-6 {
  color: #aaa;
  font-size: 0.8125rem;
  padding: 2rem 0;
}

@media (min-width:768px) {
  .emotion-6 {
    font-size: 1rem;
  }
}

.emotion-5 {
  margin: 0 auto;
  max-width: 1080px;
  padding: 0 24px;
}

.emotion-0 {
  margin-bottom: 16px;
}

.emotion-0 span {
  display: block;
}

.emotion-1 {
  font-weight: 700;
  margin-right: 0.25rem;
}

<footer
  className="emotion-6"
>
  <div
    className="emotion-5"
  >
    <div
      className="emotion-0"
    >
      <span>
        Registered office address:
      </span>
      <span>
        272 Bath Street
      </span>
      <span>
        Glasgow, G2 4JR
      </span>
    </div>
    <div
      className="emotion-2"
    >
      <span
        className="emotion-1"
      >
        Company number
        :
      </span>
      <span>
        SC588866
      </span>
    </div>
    <div
      className="emotion-2"
    >
      <span
        className="emotion-1"
      >
        VAT registration number
        :
      </span>
      <span>
        288078750
      </span>
    </div>
  </div>
</footer>
`;
